types/models.d.ts

```diff
@@ -183,6 +183,9 @@ declare module 'mongoose' {
     /* Cast the given POJO to the model's schema */
     castObject(obj: AnyObject, options?: { ignoreCastErrors?: boolean }): TRawDocType;
 
+    /* Apply defaults to the given document or POJO. */
+    applyDefaults(obj: AnyObject|TRawDocType): TRawDocType;
```

Comment:
Minor detail but I think having 2 separate overrides `applyDefaults(obj: TRawDocType): TRawDocType;` and `applyDefaults(obj: AnyObject): AnyObject` would be more correct. You can technically pass any object into `applyDefaults()`, even one that doesn't match the schema, and it will still apply defaults.

Reply:
Thanks for the suggestions. i've changed it accordingly